### ISSUE TITLE
Running LoginAsync will cause the bot to just end

### ISSUE THREAD

**Issue by MatthewNordstrand** [RogueException/Discord.Net#614]
In my code during the startup, as soon as I get to the line:

`await client.LoginAsync(TokenType.Bot, botToken);`

The bot will just close with the exit code:

`The program '[14248] StatBotN.exe' has exited with code 0 (0x0).`

without providing me any useful information on why this is happeneing. I am using the correct bot token as well. I've checked multiple times.

**Comment by foxbot** [comment]
Any information on the platform? Windows or Linux? .NET Core? 

Also include a sample of your `Program.cs` please

**Comment by foxbot** [comment]
There hasn't been any activity on this issue in over a week, so I'm going to assume you've resolved it. Feel free to reopen if you have further questions.

**Comment by Lightfire228** [comment]
So, this issue is really old, but I ran into this problem myself and took hours to figure out what was wrong.  

I was following [this example on github](https://github.com/Aux/Discord.Net-Example).  Specifically, I had moved my bot init code from `Program.Main()` to `Startup.RunAsync()` as shown [here](https://github.com/Aux/Discord.Net-Example/blob/2.0/src/Program.cs).  However, I forgot to change the return type of `Main()`  from `void` 

```cs
public static void Main(string[] args) => 
    Startup.RunAsAsync(args)
;
```
to `Task`
```cs
public static Task Main(string[] args) => 
    Startup.RunAsAsync(args)
;
```
This was causing my code to exit early, at the `await _discord.LoginAsync(TokenType.Bot, discordToken);`
